src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php

```diff
@@ -48,13 +49,20 @@ public function get(RequestConfiguration $requestConfiguration, RepositoryInterf
     {
         $resources = $this->resourcesResolver->getResources($requestConfiguration, $repository);
 
-        if ($resources instanceof Pagerfanta) {
+        if ($resources instanceof ResourceGridView or $resources instanceof Pagerfanta) {
             $request = $requestConfiguration->getRequest();
-            $resources->setMaxPerPage($requestConfiguration->getPaginationMaxPerPage());
-            $resources->setCurrentPage($request->query->get('page', 1));
+
+            if ($resources instanceof ResourceGridView) {
+                $data = $resources->getData();
+            } else {
+                $data = $resources;
+            }
```

Comment:
yes, i'ts more concise.
What do you think about renaming $data into $paginator ?